sqlitecluster/SQLiteNode.cpp

```diff
@@ -571,9 +552,11 @@ bool SQLiteNode::update() {
         SASSERTWARN(!_syncPeer);
         SASSERTWARN(!_leadPeer);
         SASSERTWARN(_db.getUncommittedHash().empty());
-        // If we're trying to shut down, just do nothing
-        if (shutdownComplete())
+        // If we're trying to shut down, just do nothing, especially don't jump directly to leading and get stuck in an
+        // endless loop.
+        if (_gracefulShutdown()) {
```

Comment:
Why change this from `shutdownComplete()` to `_gracefulShutdown()`?

Reply:
`shutdownComplete()` is public and thus locks, and our convention for this file is that public const functions lock_shared, public non-const functions lock_unique, and private functions do not lock (and thus can't call public functions).